### DISCUSSION TITLE
is it possible to fetch document from web?

### DISCUSSION THREAD

**Discussion by dickens88** [open-webui/open-webui#1680] [Ideas]
Dears,
I have 2 questions about this projects,

- 1. I see that the Documents module can upload documents to RAG's database. I wonder if it's possible to directly enter a url to grab data from a web page and enter it into RAG? I noticed that the backend code seems to have this logic, but I don't see it on the frontend page.

- 2. Will Agent invoking be supported in the future? That is, configure several different agents at the front end. Each agent has its own role. Then, these agents cooperate with each other to complete tasks, such as searching for information and completing reports.

**Comment by justinh-rahb** [comment]
URLs can be used for RAG by typing # followed by the full URL of the page you want to ingest and then selecting it from the popup menu:
https://docs.openwebui.com/tutorial/rag

Agents aren't a feature yet, but there seems to be at least one and possibly more people that have claimed to be working on PRs for this. We'll take whichever is submitted and found to best match our goals.

**Reply by UserName-wang** [reply]
But how can we fetch multiple pages based on one web url recursively? For example deep in 2 layer?

**Comment by justinh-rahb** [comment]
I'm going to move this to discussion as it does not seem to be a bug report per se, and the agent feature has been requested many times and is already on our roadmap.
